torch/distributions/categorical.py

```diff
@@ -93,11 +93,10 @@ def sample(self, sample_shape=torch.Size()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        value_shape = torch._C._infer_size(value.size(), self.batch_shape) if self.batch_shape else value.size()
```

Comment:
I don't know much about distributions, but... does this change semantics when self.batch_shape is not `None`?

Reply:
Note that `self.batch_shape` is always a `torch.Size`. This line broadcasts `value` up to a batch of params, so that a single value can be scored against multiple params.